lib/devise/models/confirmable.rb

```diff
@@ -169,6 +169,7 @@ def skip_reconfirmation!
         # in models to map to a nice sign up e-mail.
         def send_on_create_confirmation_instructions
           send_confirmation_instructions
+          skip_reconfirmation!
```

Comment:
@lmduc if there already had been this testcase you mentioned, and it hadn't been failed before your PR, so is not an example, that people faces same problem like you do.

Why do you think it should not send mail when updating user right after create?